### PR TITLE
Task: use slices for query runner

### LINKED ISSUE #3202: Replace query actions, action creators and reducers with RTK slices

### PR DESCRIPTION
**Pull request by thewahome**
## Overview

Closes #3202

### CODE REVIEW

File: src/app/services/actions/query-action-creator-util.ts [hunk 1]
```diff
@@ -230,11 +230,11 @@ async function tryParseJson(textValue: string) {
 }
 
 export function parseResponse(
-  response: any,
-  respHeaders: any = {}
+  response: Response,
+  respHeaders: { [key: string]: string } = {}
```
Comment (musale) [inline src/app/services/actions/query-action-creator-util.ts:234]:
```suggestion
  respHeaders: Record<string, string> = {}
```

File: src/app/services/slices/graphResponse.slice.ts [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+      });
+  }
+});
+
+export const { setQueryResponse } = querySlice.actions;
+export default querySlice.reducer;
+
+async function processResponse(
+  response: Response,
+  respHeaders: { [key: string]: string },
```
Comment (musale) [inline src/app/services/slices/graphResponse.slice.ts:90]:
```suggestion
  respHeaders: Record<string, string>,
```

File: src/types/query-response.ts [hunk 3]
```diff
@@ -17,5 +17,5 @@ export interface IQueryResponseProps {
 
 export interface IGraphResponse {
   body: any | undefined;
-  headers: any | undefined;
+  headers: { [key: string]: string } | undefined;
```
Comment (musale) [inline src/types/query-response.ts:20]:
```suggestion
  headers: Record<string, string> | undefined;
```

**Comment by thewahome** [comment]
Thanks for the suggestion @musale 
I've taken some time to read about the `Record<string, string>` type and seen that it's best when working with a strict, predefined set of string keys, and `{ [key: string]: string }` when more flexibility with arbitrary string keys is needed. In this situation the headers are dynamic which favours the original typing

**Review by musale — APPROVED** [review]

**Pull request merged**